app/src/main/java/org/dhis2/data/forms/dataentry/DataEntryAdapter.java

```diff
@@ -316,23 +339,33 @@ public boolean hasError() {
         return hasError;
     }
 
-    public String getErrorFieldNames() {
-        StringBuilder errorFieldNames = new StringBuilder();
-        for (FieldViewModel fieldViewModel : viewModels) {
-            if (fieldViewModel.error() != null) {
-                if (errorFieldNames.length() == 0)
-                    errorFieldNames.append(fieldViewModel.label());
-                else
-                    errorFieldNames.append(", ").append(fieldViewModel.label());
-            }
-        }
-        return errorFieldNames.toString();
-    }
-
     public void setLastFocusItem(String lastFocusItem) {
         currentFocusUid.setValue(lastFocusItem);
         this.nextFocusPosition = -1;
         this.lastFocusItem = lastFocusItem;
     }
 
+    public int getItemSpan(int position) {
+
+        if (getItemViewType(position) == SECTION || rendering == null) {
+            return 2;
+        } else {
+            switch (ProgramStageSectionRenderingType.valueOf(rendering)) {
+                case MATRIX:
+                    return 1;
+                case LISTING:
+                default:
+                case SEQUENTIAL:
+                    return 2;
```

Comment:
It looks a little weird to have the default case in the middle of other cases and not at the end.